### ISSUE TITLE
SyntaxError: Unexpected token <' to include 'ENOTFOUND

### ISSUE THREAD

**Issue by hutson** [xeb/sonar-coverage-badge#1]
Tried running `npm run test`.

``` bash
> sonar-coverage-badge@0.1.0 test /home/hutson/Workspace/sonar-coverage-badge
> mocha --reporter spec



  Badger
    ✓ should generate a success badge
    ✓ should default to gray when negative value
    ✓ should get code coverage value (272ms)
    ✓ should get code coverage value for SSL (304ms)
    1) should return parsed error
    ✓ should handle an error
    ✓ should not call success when metric does not exist (200ms)

  Index 
    ✓ should read from process.argv
    ✓ should defaults if process.argv is bad
    ✓ should defaults if process.argv is false
    ✓ should default port to 8087
    ✓ should start a server on specific port
    ✓ should return svg MIME type (251ms)
    ✓ should return cache control no-cache (171ms)
    ✓ should return new etag each time (337ms)
    ✓ should return 106x20 image (296ms)
    ✓ should return 500 status code when error (299ms)


  16 passing (2s)
  1 failing

  1) Badger should return parsed error:
     Uncaught AssertionError: expected 'SyntaxError: Unexpected token <' to include 'ENOTFOUND'
      at test/badger.test.js:50:16
      at IncomingMessage.<anonymous> (src/lib/badger.js:39:9)
      at _stream_readable.js:908:16



npm ERR! Test failed.  See above for more details.
```

``` bash
npm --version
3.3.7
```

``` bash
node --version
v0.12.7
```

**Comment by xeb** [comment]
Mind retrying?  I suspect this is related to Issue #8

**Comment by hutson** [comment]
@xeb I reran the test suite, and as you noted, it's related to the way the tests depend on a network resource, but the issue was not solved by the resolution to #8.

I ran `node-inspector` to see what was causing the `parse error` for the _should return parsed error_ test, and what I found was that the badger code was trying to fetch coverage data from the dns site `http://fdasfdsfasdfdsfasdfsd`, which doesn't exist on my network. Instead, what the badger code received in response was the following:

``` html
<html><head><meta http-equiv="refresh" content="0;url=http://www.dnsrsearch.com/index.php?origURL=http://fdasfdsfasdfdsfasdfsd/api/resources&bc="/></head><body><script type="text/javascript">window.location="http://www.dnsrsearch.com/index.php?origURL="+escape(window.location)+"&r="+escape(document.referrer)+"&bc=";</script></body></html>
```

The badger code then calls `JSON.parse` on that response, which fails, since the response is not valid JSON. 

Below is a view within `node-inspector`:

![parse-error](https://cloud.githubusercontent.com/assets/6701030/11017130/4b28209a-855b-11e5-9d08-fc862b4d37f1.png)
